language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/ConfigTomlValueGenerationUtil.java

```diff
@@ -0,0 +1,163 @@
+            }
+            case STRING, XML, ANYDATA, JSON, SINGLETON -> {
+                return new TomlEntryValue("\"\"", true);
+            }
+            case TUPLE -> {
+                return new TomlEntryValue("[]", true);
+            }
+            case ARRAY -> {
+                ArrayTypeSymbol arrayTypeSymbol = (ArrayTypeSymbol) type;
+                TypeSymbol memberType = CommonUtil.getRawType(arrayTypeSymbol.memberTypeDescriptor());
+                if (memberType.typeKind() == TypeDescKind.UNION) {
+                    memberType = ((UnionTypeSymbol) memberType).memberTypeDescriptors().get(0);
+                }
+                if (memberType.subtypeOf(basicType) || basicType.subtypeOf(anydataOrJson)) {
+                    return new TomlEntryValue("[]", true);
+                }
+                return new TomlEntryValue(String.format("[[%s]]%n", confName), false); // records and maps
+            }
+            case MAP -> {
+                return new TomlEntryValue(String.format("[%s]%n", confName), false);
+            }
+            case RECORD -> {
+                StringBuilder stringBuilder = new StringBuilder();
+                stringBuilder.append(String.format("[%s]%n", confName));
+                getRecordFields((RecordTypeSymbol) type, stringBuilder, "");
+                return new TomlEntryValue(stringBuilder.toString(), false);
+            }
+            case TABLE -> {
+                return new TomlEntryValue(String.format("[[%s]]%n", confName), false);
+            }
+            case UNION -> {
+                return  getDefaultValueStr(((UnionTypeSymbol) type).memberTypeDescriptors().get(0),
+                        basicType, anydataOrJson, confName);
+            }
+            case TYPE_REFERENCE -> {
+                return getDefaultValueStr(CommonUtil.getRawType(type), basicType, anydataOrJson, confName);
+            }
+            default -> {
+                return new TomlEntryValue("", true);
+            }
+        }
+    }
+
+    private static StringBuilder getRecordFields(RecordTypeSymbol recordTypeSymbol, StringBuilder
+            builder, String prefix) {
+        List<RawTypeSymbolWrapper<RecordTypeSymbol>> recordTypeSymbols =
+                RecordUtil.getRecordTypeSymbols(recordTypeSymbol);
+
+        List<RecordFieldSymbol> validFields = new ArrayList<>();
+        Map<String, RecordFieldSymbol> fields;
+        for (RawTypeSymbolWrapper<RecordTypeSymbol> symbol : recordTypeSymbols) {
+            fields = RecordUtil.getRecordFields(symbol, Collections.emptyList());
+            validFields.addAll(fields.values());
+        }
+        for (RecordFieldSymbol recordFieldSymbol : validFields) {
+            TypeSymbol typeSymbol = CommonUtil.getRawType(recordFieldSymbol.typeDescriptor());
+            if (typeSymbol.typeKind() == TypeDescKind.RECORD) {
+                String nestedPrefix = prefix + recordFieldSymbol.getName().get() + ".";
```

Comment:
Don't we need to check if the name of the record field symbol is present?

Reply:
There cannot be any scenario that there won't be a name for a field. Field symbols exists because there is name associated to it. Thats why I didn't check it in this particular case.